racket/collects/net/url-string.rkt

```diff
@@ -176,17 +176,20 @@
            "(?:"              ; | / user-at-opt
```

Comment:
I've actually found the pointed-to appendix (https://tools.ietf.org/html/rfc3986#appendix-B), and learned that it does support IPv6. In fact, that regexp uses `([^/?#]*)` for the host part; the custom adaption here `([^/?#:]*)` breaks that support by excluding `:`.